jerry-core/ecma/operations/ecma-regexp-object.c

```diff
@@ -1434,10 +1434,13 @@ ecma_regexp_exec_helper (ecma_value_t regexp_value, /**< RegExp object */
           capture_value = ecma_make_string_value (capture_str_p);
         }
 
+        ecma_property_descriptor_t array_item_prop_desc = ecma_make_empty_property_descriptor ();
+        array_item_prop_desc.is_value_defined = true;
+        array_item_prop_desc.value = capture_value;
         ECMA_TRY_CATCH (put_value,
-                        ecma_op_object_put (result_array_obj_p,
+                        ecma_op_object_define_own_property (result_array_obj_p,
                                             index_str_p,
-                                            capture_value,
+                                            &array_item_prop_desc,
                                             true),
```

Comment:
Oh, it is even more easier then. Just use ecma_create_named_data_property / ecma_set_named_data_property_value. You can even delete the ecma_free_value (capture_value)